### DISCUSSION TITLE
Can I call a template depending on a variable like `.chezmoi.osRelease.id`?

### DISCUSSION THREAD

**Discussion by lwbt** [twpayne/chezmoi#3522] [Q&A] [answered]
I have the following code:

```
{{- if eq .chezmoi.osRelease.id "ubuntu" -}}
{{-   template "cz_templates/bashrc_ubuntu" . -}}
{{- else if eq .chezmoi.osRelease.id "steamos" -}}
{{-   template "cz_templates/bashrc_steamos" . -}}
{{- end -}}
```

Would it instead be possible to skip the lengthy if ... else and write code that comes close to this?

```
{{- template "cz_templates/bashrc_" .chezmoi.osRelease.id . -}}
```

I tried `join "bashrc_" .chezmoi.osRelease.id` and `(printf "bashrc_" .chezmoi.osRelease.id)`. I tried looking for similar topics related to go and templating and looking at https://pkg.go.dev/text/template, but I can't figure it out. Unfortunately I don't write as much go and template code as I want to in my day job and my little go knowledge from a few years ago is quite rusty.

**Comment by 2-4601** [comment, marked as answer]
You could try [`includeTemplate`](https://www.chezmoi.io/reference/templates/functions/includeTemplate/).

```
{{- includeTemplate (print "cz_templates/bashrc_" .chezmoi.osRelease.id) . -}}
```
